core/src/cost_update_service.rs

```diff
@@ -174,6 +190,36 @@ impl CostUpdateService {
         );
         dirty
     }
+
+    // Based historical data, it seems reasonable to flag it as large change if updated_cost
+    // is more than double of input_data;
+    // Compare updated_cost against input_cost instead of previous calculated cost captures
+    // how far apart between single data point and calculated value, which is a good indicator
+    // of potential issues.
+    fn is_large_change_in_cost_calculation(input_cost: i64, updated_cost: i64) -> bool {
+        (updated_cost - input_cost).abs() > input_cost
+    }
+
+    fn report_large_change_in_cost(
+        program_id: &Pubkey,
+        program_timing: &ProgramTiming,
+        calculated_units: u64,
+    ) {
+        datapoint_info!(
```

Comment:
over what period are these value aggregated?

Reply:
It is not aggregated. I'm thinking to capture each potential bad calc, easier for investigation (do paper and pencil math).

Reply:
I think that's fine so we can confidently throttle submissions.  Don't want to knock influx over :sweat_smile: